File: cucumber-mockup/cnti-tests/features/lib/step_definitions/workload/configuration.rb
Given ("CNF is deployed in dry mode") do
  if !CNFManager.cnf_installed? do
    Log.info("Installing CNF in dry-run from #{Config.cnf_location} file")
    CNFManager.cnf_install(params:['--dry-run'])
    Log.info("CNF installed")
  else do
    Log.info("CNF is already installed")
  end
end

Then ("Helm files do not contain hardcoded ip addresses") do
  cnf_installation_dir = config['cnf_installation_dir']
  chart_path = "#{cnf_installation_dir}/helm_chart.yml"

  File.foreach(chart_path).with_index do |line, line_number|
    next if line =~ /NOTES:/
    match = line.match("/([0-9]{1,3}[\.]){3}[0-9]{1,3}/")
    if match[0] != "0.0.0.0" && match[0] != "127.0.0.1" do
      raise "Runtime Helm configuration at #{chart_path} contains hardcoded ip address on line #{line_number}"
    file_content = File.read(file_path)
    expect(file_content).not_to match(ip_regex), "File #{file_path} contains a fixed IP address."
  end
end

File: cucumber-mockup/cnti-tests/features/lib/step_definitions/workload/configuration.py
from behave import given, then
import re
import os


@given("CNF is deployed in dry mode")
def step_given_cnf_deployed_in_dry_mode(context):
    if not CNFManager.cnf_installed():
        logging.info(f"Installing CNF in dry-run from {context.config.cnf_location} file")
        CNFManager.cnf_install(params=['--dry-run'])
        logging.info("CNF installed")
    else:
        logging.info("CNF is already installed")

@then("Helm files do not contain hardcoded ip addresses")
def step_then_helm_files_do_not_contain_hardcoded_ip_addresses(context):
    cnf_installation_dir = context.config['cnf_installation_dir']
    chart_path = os.path.join(cnf_installation_dir, "helm_chart.yml")

    ip_regex = re.compile(r"\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b")
    with open(chart_path, 'r') as file:
        for line_number, line in enumerate(file, start=1):
            if "NOTES:" in line:
                continue
            match = ip_regex.search(line)
            if match and match.group(0) not in ["0.0.0.0", "127.0.0.1"]:
                raise Exception("Runtime Helm configuration at {} contains hardcoded ip address on line {}".format(chart_path, line_number))
